src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: LGPL-3.0-or-newer
+pragma solidity ^0.6.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/// @title Gnosis Protocol v2 Encoding Library.
+/// @author Gnosis Developers
+library GPv2Encoding {
+    /// @dev A struct representing an executed order that can be used for
+    /// settling a batch.
+    struct Order {
+        address owner;
+        IERC20 sellToken;
+        IERC20 buyToken;
+        uint256 sellAmount;
+        uint256 buyAmount;
+        uint32 validTo;
+        uint32 nonce;
+        uint256 tip;
+        OrderKind kind;
+        bool partiallyFillable;
+        uint256 executedAmount;
+        uint8 sellTokenIndex;
+        uint8 buyTokenIndex;
+        bytes32 digest;
+    }
+
+    /// @dev An enum describing an order kind, either a buy or a sell order.
+    enum OrderKind {Sell, Buy}
+
+    /// @dev The stride of an encoded order.
+    uint256 internal constant ORDER_STRIDE = 204;
+
+    /// @dev Bit position for the [`OrderKind`] encoded in the flags.
+    uint8 internal constant ORDER_KIND_BIT = 0;
+    /// @dev Bit position for the `partiallyFillable` value encoded in the
+    /// flags.
+    uint8 internal constant ORDER_PARTIALLY_FILLABLE_BIT = 1;
+
+    /// @dev Decodes a signed order from calldata bytes into memory.
+    /// Orders are tightly packed in order to reduce calldata and associated gas
+    /// costs. They contain the following fields:
+    /// ```
+    /// struct EncodedOrder {
+    ///     uint8 sellTokenIndex;
+    ///     uint8 buyTokenIndex;
+    ///     uint256 sellAmount;
+    ///     uint256 buyAmount;
+    ///     uint32 validTo;
+    ///     uint32 nonce;
+    ///     uint256 tip;
+    ///     uint8 flags;
+    ///     uint256 executedAmount;
+    ///     Signature {
+    ///         uint8 v;
+    ///         bytes32 r;
+    ///         bytes32 s;
+    ///     } signature;
+    /// }
+    /// ```
+    /// @param tokens The list of tokens included in the settlement. The encoded
```

Comment:
Was leaving out `domainSeparator` intentional?

Reply:
No, it was not.